test/units/module_utils/basic/test_log.py

```diff
@@ -0,0 +1,227 @@
+    def test_smoketest_journal(self):
+        # These talk to the live daemons on the system.  Need to do this to
+        # show that what we send doesn't cause an issue once it gets to the
+        # daemon.  These are just smoketests to test that we don't fail.
+
+        self.am.log(u'Text string')
+        self.am.log(u'Toshio くらとみ non-ascii test')
+
+        self.am.log(b'Byte string')
+        self.am.log(u'Toshio くらとみ non-ascii test'.encode('utf-8'))
+        self.am.log(b'non-utf8 :\xff: test')
+
+
+class TestAnsibleModuleLogSyslog(unittest.TestCase):
+    """Test the AnsibleModule Log Method"""
+
+    py2_output_data = {
+            u'Text string': b'Text string',
+            u'Toshio くらとみ non-ascii test': u'Toshio くらとみ non-ascii test'.encode('utf-8'),
+            b'Byte string': b'Byte string',
+            u'Toshio くらとみ non-ascii test'.encode('utf-8'): u'Toshio くらとみ non-ascii test'.encode('utf-8'),
+            b'non-utf8 :\xff: test': b'non-utf8 :\xff: test'
+            }
+
+    py3_output_data = {
+            u'Text string': u'Text string',
+            u'Toshio くらとみ non-ascii test': u'Toshio くらとみ non-ascii test',
+            b'Byte string': u'Byte string',
+            u'Toshio くらとみ non-ascii test'.encode('utf-8'): u'Toshio くらとみ non-ascii test',
+            b'non-utf8 :\xff: test': b'non-utf8 :\xff: test'.decode('utf-8', 'replace')
+            }
+
+    def setUp(self):
+        self.complex_args_token = basic.MODULE_COMPLEX_ARGS
+        basic.MODULE_COMPLEX_ARGS = '{}'
+        self.am = basic.AnsibleModule(
+            argument_spec = dict(),
+        )
+        self.has_journal = basic.has_journal
+        if self.has_journal:
+            # Systems with journal can still test syslog
+            basic.has_journal = False
+
+    def tearDown(self):
+        basic.MODULE_COMPLEX_ARGS = self.complex_args_token
+        basic.has_journal = self.has_journal
+
+    @patch('syslog.syslog', autospec=True)
+    def test_no_log(self, mock_func):
+        no_log = self.am.no_log
+
+        self.am.no_log = True
+        self.am.log('unittest no_log')
+        self.assertFalse(mock_func.called)
+
+        self.am.no_log = False
+        self.am.log('unittest no_log')
+        mock_func.assert_called_once_with(syslog.LOG_INFO, 'unittest no_log')
+
+        self.am.no_log = no_log
+
+    def test_output_matches(self):
+        if sys.version_info >= (3,):
+            for msg, param in self.py3_output_data.items():
+                with patch('syslog.syslog', autospec=True) as mock_func:
+                    self.am.log(msg)
+                    mock_func.assert_called_once_with(syslog.LOG_INFO, param)
+        else:
+            for msg, param in self.py2_output_data.items():
+                with patch('syslog.syslog', autospec=True) as mock_func:
+                    self.am.log(msg)
+                    mock_func.assert_called_once_with(syslog.LOG_INFO, param)
```

Comment:
The code is the same, just the data differs, so I think it would be clearer to move the loop outside the if.
```python
output_data = self.py3_output_data if sys.version_info[0] >= 3 else self.py2_output_data
for msg, param in output_data.items():
    ...
```